### PR TITLE
Add compatibility for 0.11, 0.12 and iojs

### PR DESCRIPTION
**Pull request by pksunkara**
For @zdne to review and merge.

Tested locally on nodejs 0.8, 0.10, 0.11, 0.12 and iojs 1.2

### CODE REVIEW

**Comment by pksunkara** [comment]
Should fix #16 /cc @Almad

**Comment by pksunkara** [comment]
iojs tests errored out on travis because node-gyp was unable to download iojs tarball.

**Comment by zdne** [comment]
@pksunkara what you mean by "We should not update npm" that we shouldn't release new version of the nom package?

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+  - "0.12"
+  - "0.11"
   - "0.10"
   - "0.8"
+  - iojs
 script: "npm install && npm test"
 before_install:
   - git submodule update --init --recursive
-  - npm install -g npm@~1.4.6
+  - ruby -e "if '`npm -v`'<'2.0.0' then \`npm i -g npm\` else '' end"
```
Comment (zdne) [inline .travis.yml]: ruby?

**Comment by pksunkara** [comment]
> what you mean by "We should not update npm" that we shouldn't release new version of the nom package?

No, I meant that we should not update the version of npm which was installed by the io.js

File: .travis.yml (hunk 1 above)
Comment (pksunkara) [inline .travis.yml]: I would have used shell or bash if I know how to do this. :disappointed:

Comment (kylef) [inline .travis.yml]: npm can install npm?

mind blown.

Comment (zdne) [inline .travis.yml]: So if I understand it correctly Npm > 2.0 is required?

Comment (kylef) [inline .travis.yml]: This ruby logic can be simplified:

```diff
- ruby -e "if '`npm -v`'<'2.0.0' then \`npm i -g npm\` else '' end"
+ ruby -e "`npm i -g npm` if `npm -v` < '2.0.0'"
```

Comment (zdne) [inline .travis.yml]: :+1:

Comment (pksunkara) [inline .travis.yml]: > So if I understand it correctly Npm > 2.0 is required?

Well, npm should be atleast 1.4.6. But I have trouble trying to do `<` and `>` operations on it. So, I used `2.0.0` which was easier to do those operations. Except `0.8` and `0.10`, all other versions have npm > 2.0.0

**Comment by pksunkara** [comment]
@zdne Updated the readme.

**Pull request merged**